analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/CollectionTracker.cs

```diff
@@ -81,10 +96,27 @@ public static CollectionConstraint ObjectCreationConstraint(ProgramState state,
             ? CollectionConstraint.Empty
             : CollectionConstraint.NotEmpty;
 
-    public static CollectionConstraint MethodReferenceConstraint(IMethodReferenceOperationWrapper operation) =>
-        operation.Instance is not null && AddMethods.Contains(operation.Method.Name)
-            ? CollectionConstraint.NotEmpty
-            : null;
+    public static ProgramState LearnFrom(ProgramState state, IMethodReferenceOperationWrapper operation)
+    {
+        if (operation.Instance is not null
+            && operation.Instance.Type.DerivesOrImplementsAny(CollectionTypes)
+            && AddOrRemoveMethods.Contains(operation.Method.Name))
+        {
+            state = state[operation.Instance] is { } operationValue
+                ? state.SetOperationValue(operation, operationValue.WithoutConstraint<CollectionConstraint>())
+                : state;
+
+            state = operation.Instance.TrackedSymbol(state) is { } symbol && state[symbol] is { } symbolValue
```

Comment:
We don't need to look up `state[symbol]`. Just set the operation and symbol to the same `SymbolicValue`. The assumption is that they have the same value anyway.

Reply:
Good point.
Also, I was setting the `operation` above, instead of the `operation.Instance`

Reply:
hmm... that's an indication that UT coverage is sub-optimal. Please take a look and see if it makes sense to add one or two UTs.